thorin/world.h

```diff
@@ -145,10 +150,23 @@ class World {
         return lam;
     }
     Lam* nom_lam(const Pi* cn, const Def* dbg = {}) { return nom_lam(cn, Lam::CC::C, dbg); }
+
+    Lam* nom_filter_lam(const Pi* cn, const Def* dbg){
+        return nom_filter_lam(cn, lit_true(), dbg);
+    }
+
+    Lam* nom_filter_lam(const Pi* cn, const Def* filter, const Def* dbg){
+        Lam* lam = nom_lam(cn, dbg);
+        lam->set_filter(filter);
+        return lam;
+    }
+
     const Lam* lam(const Pi* pi, const Def* filter, const Def* body, const Def* dbg) {
         return unify<Lam>(2, pi, filter, body, dbg);
     }
     const Lam* lam(const Pi* pi, const Def* body, const Def* dbg) { return lam(pi, lit_true(), body, dbg); }
+    const Lam* flatten_lam(Lam* lam);
+    const Lam* unflatten_lam(Lam* lam);
     ///@}
```

Comment:
I think we should move these helper functions to somewhere else. `World` should just have factory methods (smart constructors). We can either move that to `tuple.*` which already contains some flatten/unflatten helper functions or to `lam.*`.